llvm/lib/Target/SPIRV/SPIRVSubtarget.h

```diff
@@ -56,24 +53,20 @@ class SPIRVSubtarget : public SPIRVGenSubtargetInfo {
   bool OpenCLFullProfile;
   bool OpenCLImageSupport;
 
-  SPIRVInstrInfo InstrInfo;
-  SPIRVFrameLowering FrameLowering;
-  SPIRVTargetLowering TLInfo;
+  std::set<Extension::Extension> AvailableExtensions;
+  std::set<ExtInstSet> AvailableExtInstSets;
+  std::set<Capability::Capability> AvailableCaps;
 
-  // TODO Some of these fields might work without unique_ptr.
-  //      But they are shared with other classes, so if the SPIRVSubtarget
-  //      moves, not relying on unique_ptr breaks things.
   std::unique_ptr<SPIRVGeneralDuplicatesTracker> DT;
   std::unique_ptr<SPIRVGlobalRegistry> GR;
-  std::unique_ptr<SPIRVCallLowering> CallLoweringInfo;
-  std::unique_ptr<SPIRVRegisterBankInfo> RegBankInfo;
 
-  std::unordered_set<Extension::Extension> AvailableExtensions;
-  std::unordered_set<ExtInstSet> AvailableExtInstSets;
-  std::unordered_set<Capability::Capability> AvailableCaps;
```

Comment:
Why are we doing this? I haven't see any comment about this on Phabricator

Reply:
Yes, this fragment is not on review. However std::unordered_set is not used in LLVM project (see https://llvm.org/docs/ProgrammersManual.html), so we need to get rid of it anyway. Perhaps std::set is not the best candidate for this case and we should use another set implementation e.g. SmallSet.